File: api/app/model/room.go
package model

import "time"

type RoomID uint64

type Room struct {
	ID        RoomID
	Name      string
	IsGroup   bool
	CreatedAt time.Time
	UpdatedAt time.Time
	Img       *string
}

type RoomUser struct {
	Room  Room
	Users []*ShowUser
}

type IndexRoom struct {
	Room      Room
	IsOpen    bool
	LastText  *string
	CountUser int
	ShowImg   *string
}

File: client/lib/model/room.ts
import { User } from "./user";

export class Room {
  constructor(
    public id: number,
    public name: string,
    public isGroup: boolean,
    public createdAt: Date,
    public updatedAt: Date,
    public img?: string,
  ) {}
}

export interface ShowRoom {
  room: Room;
  users: Omit<User, "email" | "password" | "createdAt">[];
}

export interface IndexRoom {
  room: Room;
  isOpen: boolean;
  countUser: number;
  lastText?: string;
  showImg?: string;
}

export interface AllRoom {
  indexRooms: IndexRoom[];
  nextID: number | null;
}
